modules/tinymce/src/core/main/ts/DragDropOverrides.ts

```diff
@@ -121,6 +131,54 @@ const moveGhost = (
 
   ghostElm.style.width = (width - overflowX) + 'px';
   ghostElm.style.height = (height - overflowY) + 'px';
+
+  // Code needed for dragging CEF elements (specifically fixing TINY-8874)
+  // The idea behind the algorithm is that the user will start dragging the
+  // CEF element to the edge of the editor and that would cause scrolling.
+  // The way that happens is that the user will trigger a mousedown event,
+  // then a mousemove event until they reach the edge of the editor. Then
+  // no event triggers. That's when I set an interval to keep scrolling the editor.
+  // Once a new event triggers I clear the existing interval and set it back to none.
+
+  const clientHeight = contentAreaContainer.clientHeight;
+
+  // current scroll level
+  const currentTop = win.scrollY;
+
+  state.on((state) => {
+    if (state.dragging) {
+      // This basically means that the mouse is close to the bottom edge
+      // (within MouseRange pixels of the bottom edge)
+      if (mouseY + MouseRange >= clientHeight) {
+        const scrollDown = (currentTop: number) => {
+          win.scroll({
+            top: currentTop + ScrollingValue,
+            behavior: 'smooth'
+          });
+        };
+        scrollDown(currentTop);
+        state.intervalId = Optional.some(setInterval(() => {
```

Comment:
Wonder whether this can be optimized further by clearing the previous timer before setting a new one

Reply:
I do clear previous timers before calling `moveGhost` [here](https://github.com/tinymce/tinymce/blob/1cb5ddce8eac335b535725a19355f81ad4a8a78f/modules/tinymce/src/core/main/ts/DragDropOverrides.ts#L249)